### ISSUE TITLE
Got 'Could not convert glyph to bitmap' error when run matplotlib fig.savefig() in parallel (mp.Pool)

### ISSUE THREAD

**Issue by Yongzhi-GenapSys** [matplotlib/matplotlib#14524]
### Bug report

**Bug summary**
Got 'Could not convert glyph to bitmap' error when run matplotlib fig.savefig() in parallel 

Recently we got this 'Could not convert glyph to bitmap' error from time to time after we implemented a multi-process solution of the pipeline. That is, there could be multiple processes plotting and saving the figures simultaneously. We had removed all individual fig.savefig() invoking at the top level and forced to use util function in every place. Now the error tracks back to that last fig.savefig() within util function which was pretty bad. It seemed that the more complicated the figure, the more processes running at the same time, the higher possibility we got this issue. 

**Code for reproduction**

```python
# Paste your code here
# We used the following line to save a figure in utli.save_figures()
fig.savefig(full_path, pad_inches=pad_inches, dpi=dpi)
```

**Actual outcome**

```
# If applicable, paste the console output here
   File "analysis/util.py", line 1041, in save_figures
    fig.savefig(full_path, pad_inches=pad_inches, dpi=dpi)
  File "/root/eureka1.0/local/lib/python2.7/site-packages/matplotlib/figure.py", line 2062, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "/root/eureka1.0/local/lib/python2.7/site-packages/matplotlib/backend_bases.py", line 2263, in print_figure
    **kwargs)
  File "/root/eureka1.0/local/lib/python2.7/site-packages/matplotlib/backends/backend_agg.py", line 517, in print_png
    FigureCanvasAgg.draw(self)
  File "/root/eureka1.0/local/lib/python2.7/site-packages/matplotlib/backends/backend_agg.py", line 437, in draw
    self.figure.draw(self.renderer)
  File "/root/eureka1.0/local/lib/python2.7/site-packages/matplotlib/artist.py", line 55, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/root/eureka1.0/local/lib/python2.7/site-packages/matplotlib/figure.py", line 1493, in draw
    renderer, self, artists, self.suppressComposite)
  File "/root/eureka1.0/local/lib/python2.7/site-packages/matplotlib/image.py", line 141, in _draw_list_compositing_images
    a.draw(renderer)
  File "/root/eureka1.0/local/lib/python2.7/site-packages/matplotlib/artist.py", line 55, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/root/eureka1.0/local/lib/python2.7/site-packages/matplotlib/axes/_base.py", line 2635, in draw
    mimage._draw_list_compositing_images(renderer, self, artists)
  File "/root/eureka1.0/local/lib/python2.7/site-packages/matplotlib/image.py", line 141, in _draw_list_compositing_images
    a.draw(renderer)
  File "/root/eureka1.0/local/lib/python2.7/site-packages/matplotlib/artist.py", line 55, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/root/eureka1.0/local/lib/python2.7/site-packages/matplotlib/text.py", line 755, in draw
    ismath=ismath, mtext=mtext)
  File "/root/eureka1.0/local/lib/python2.7/site-packages/matplotlib/backends/backend_agg.py", line 212, in draw_text
    font.draw_glyphs_to_bitmap(antialiased=rcParams['text.antialiased'])
RuntimeError: In draw_glyphs_to_bitmap: Could not convert glyph to bitmap
```

**Expected outcome**
All figures are successfully saved.

**Matplotlib version**

  * Operating system: Ubuntu 16.04.4
  * Matplotlib version:  2.2.4
  * Matplotlib backend (`print(matplotlib.get_backend())`): agg
  * Python version:  2.7.12

**Comment by QuLogic** [comment]
I think this is a duplicate of #13723.

**Comment by tacaswell** [comment]
This issue is fixed in mpl 3.0+ (via a python 3 only API to drop the font cache on fork).  There is only going to be one more py2 compatible release and we are not going to have this bug fixed by then.  Closing as "won't or can't fix".

Sorry @Yongzhi-GenapSys .

**Comment by hugokitano** [comment]
Hi, I got an error identical to this one but with Python 3.7.9, using matplotlib 3.3.2.

Here's the code:

```
features_colors = [ colors[int(data[i,18])] for i in range(len(X)) ]
    # plot the PCA components
    label_dict = {colors[1]: 'Orbitrap-1',
            colors[2]: 'Orbitrap-2',
            colors[3]: 'Orbitrap-3'}
    labels = [label_dict[features_colors[i]] for i in range(len(features_colors))]
    plt.scatter(X_pca[:, 0], X_pca[:, 1],
                c=features_colors, marker='o',
                alpha=0.4, label=labels
            )

    # plot the centroids
    plt.title("PCA of Seer Instrument Type")
    first_component_variance = round(pca_model.explained_variance_ratio_[0] * 100, 2)
    second_component_variance = round(pca_model.explained_variance_ratio_[1] * 100, 2)
    plt.xlabel(f'First component: {first_component_variance}% of variance')
    plt.ylabel(f'Second component: {second_component_variance}% of variance')
    #plt.show()
    plt.legend()
    plt.savefig("outputs/instruments_1.0.png")

    plt.clf()
```
And the error:

```
Traceback (most recent call last):
  File "unsupervised_learning.py", line 112, in <module>
    main()
  File "unsupervised_learning.py", line 84, in main
    features_colors = [ colors[int(data[i,18])] for i in range(len(X)) ]
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/pyplot.py", line 859, in savefig
    res = fig.savefig(*args, **kwargs)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/figure.py", line 2311, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/backend_bases.py", line 2217, in print_figure
    **kwargs)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/backend_bases.py", line 1639, in wrapper
    return func(*args, **kwargs)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py", line 509, in print_png
    FigureCanvasAgg.draw(self)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py", line 407, in draw
    self.figure.draw(self.renderer)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/artist.py", line 41, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/figure.py", line 1864, in draw
    renderer, self, artists, self.suppressComposite)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/image.py", line 131, in _draw_list_compositing_images
    a.draw(renderer)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/artist.py", line 41, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/cbook/deprecation.py", line 411, in wrapper
    return func(*inner_args, **inner_kwargs)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/axes/_base.py", line 2747, in draw
    mimage._draw_list_compositing_images(renderer, self, artists)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/image.py", line 131, in _draw_list_compositing_images
    a.draw(renderer)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/artist.py", line 41, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/legend.py", line 614, in draw
    self._legend_box.draw(renderer)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/offsetbox.py", line 362, in draw
    c.draw(renderer)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/offsetbox.py", line 362, in draw
    c.draw(renderer)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/offsetbox.py", line 362, in draw
    c.draw(renderer)
  [Previous line repeated 1 more time]
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/offsetbox.py", line 913, in draw
    self._text.draw(renderer)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/artist.py", line 41, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/text.py", line 727, in draw
    ismath=ismath, mtext=mtext)
  File "/opt/anaconda3/envs/instrument-class/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py", line 203, in draw_text
    antialiased=mpl.rcParams['text.antialiased'])
RuntimeError: In draw_glyphs_to_bitmap: Could not convert glyph to bitmap (error code 0x62)
```

**Comment by jklymak** [comment]
@hugokitano please open a new report - also please include a stand-alone minimal script that triggers the error.
